src/reflect/scala/reflect/internal/util/OwnerOnlyChmod.scala

```
@@ -0,0 +1,110 @@
+      if (!f(true, true)) fail()
+    }
+    if (file.isDirectory) {
+      clearAndSetOwnerOnly(file.setExecutable)
+    }
+    clearAndSetOwnerOnly(file.setReadable)
+    clearAndSetOwnerOnly(file.setWritable)
+  }
+}
+
+
+object NioAclChmodReflective {
+  private class Reflectors {
+    val file_toPath = classOf[java.io.File].getMethod("toPath")
+    val files = Class.forName("java.nio.file.Files")
+    val path_class = Class.forName("java.nio.file.Path")
+    val getFileAttributeView = files.getMethod("getFileAttributeView", path_class, classOf[Class[_]], Class.forName("[Ljava.nio.file.LinkOption;"))
+    val linkOptionEmptyArray = java.lang.reflect.Array.newInstance(Class.forName("java.nio.file.LinkOption"), 0)
+    val aclFileAttributeView_class = Class.forName("java.nio.file.attribute.AclFileAttributeView")
+    val aclEntry_class = Class.forName("java.nio.file.attribute.AclEntry")
+    val aclEntryBuilder_class = Class.forName("java.nio.file.attribute.AclEntry$Builder")
+    val newBuilder = aclEntry_class.getMethod("newBuilder")
+    val aclEntryBuilder_build = aclEntryBuilder_class.getMethod("build")
+    val userPrinciple_class = Class.forName("java.nio.file.attribute.UserPrincipal")
+    val setPrincipal = aclEntryBuilder_class.getMethod("setPrincipal", userPrinciple_class)
+    val setPermissions = aclEntryBuilder_class.getMethod("setPermissions", Class.forName("[Ljava.nio.file.attribute.AclEntryPermission;"))
+    val aclEntryType_class = Class.forName("java.nio.file.attribute.AclEntryType")
+    val setType = aclEntryBuilder_class.getMethod("setType", aclEntryType_class)
+    val aclEntryPermission_class = Class.forName("java.nio.file.attribute.AclEntryPermission")
+    val aclEntryPermissionValues = aclEntryPermission_class.getDeclaredMethod("values")
+    val aclEntryType_ALLOW = aclEntryType_class.getDeclaredField("ALLOW")
+  }
+  private val reflectors = try { new Reflectors } catch { case ex: Throwable => null }
```

should we log the swallowed exception?